pydantic/_internal/_typing_extra.py

```diff
@@ -68,6 +69,11 @@ def origin_is_union(tp: type[Any] | None) -> bool:
 
 
 NoneType = None.__class__
+if sys.version_info < (3, 10):
+    # NoneType = type(None)
```

Comment:
I see there is `NoneType = None.__class__` above, I was thinking about changing things to handle `NoneType` the same as `EllipsisType` while I was at it since they were added at the same time/place/for the same reasons. But I guess the current definition is working? I'll drop these commented lines related to NoneType if you don't want to change it.

To be clear, I do get mypy errors if I don't use the import from `types` in python 3.11:
```
pydantic/_internal/_core_metadata.py:100: error: Variable "pydantic._internal._typing_extra.EllipsisType" is not valid as a type  [valid-type]
pydantic/_internal/_core_metadata.py:100: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
```